### DISCUSSION TITLE
editing audio plugin

### DISCUSSION THREAD

**Discussion by rosemintz** [jspsych/jsPsych#848] [Q&A]
I am trying to make a simple cosmetic edit to the audio-button-response plugin and running into an error.

I would like to make alternate versions of both html-button-response and audio-button-response so that, in certain trials, my buttons will display in two inline blocks instead of one. For the html plugin, this was easy to do by breaking up lines 96-101 of the original plugin into two loops:

```javascript
html += '<div id="jspsych-html-button-response-btngroup">';
var halflength = trial.choices.length / 2;

for (var i = 0; i < halflength; i++) {
      var str = buttons[i].replace(/%choice%/g, trial.choices[i]);
      html += '<div class="jspsych-html-button-response-button" style="display: inline-block; margin:'+trial.margin_vertical+' '+trial.margin_horizontal+'" id="jspsych-html-button-response-button-' + i +'" data-choice="'+i+'">'+str+'</div>';
    }
html += '</div> <div id="jspsych-html-button-response-btngroup">';

for (var i = halflength; i < trial.choices.length; i++){
      var str = buttons[i].replace(/%choice%/g, trial.choices[i]);
      html += '<div class="jspsych-html-button-response-button" style="display: inline-block; margin:'+trial.margin_vertical+' '+trial.margin_horizontal+'" id="jspsych-html-button-response-button-' + i +'" data-choice="'+i+'">'+str+'</div>';
    }
html += '</div>';
```

When I try to do the same in the audio-button plugin, I get the error

> TypeError: null is not an object (evaluating 'display_element.querySelector('#jspsych-audio-button-response-button-' + i).addEventListener')

I don't know enough to understand what is causing this or how to fix it, any advice would be appreciated!

**Comment by jodeleeuw** [comment]
Hi @rosemintz. Can you show the code that you are editing in the `audio-button-response` plugin?

**Reply by rosemintz** [reply]
Yes, so the main edit is replacing the loop in lines 118-121 with the following: 
```javascript
var halflength = trial.choices.length / 2;
for (var i = 0; i < halflength; i++) {
      var str = buttons[i].replace(/%choice%/g, trial.choices[i]);
      html += '<div class="jspsych-audio-button-response-button" style="display: inline-block; margin:'+trial.margin_vertical+' '+trial.margin_horizontal+'" id="jspsych-html-button-response-button-' + i +'" data-choice="'+i+'">'+str+'</div>';
    }
html += '</div> <div id="jspsych-audio-button-response-btngroup">';
for (var i = halflength; i < trial.choices.length; i++){
      var str = buttons[i].replace(/%choice%/g, trial.choices[i]);
      html += '<div class="jspsych-audio-button-response-button" style="display: inline-block; margin:'+trial.margin_vertical+' '+trial.margin_horizontal+'" id="jspsych-html-button-response-button-' + i +'" data-choice="'+i+'">'+str+'</div>';
    }
```
In addition, I've changed the name of the plugin to "audio-button-response-rows" (since I am also still using the unedited plugin in other trials) in lines 11-17: 
``` javascript
jsPsych.plugins["audio-button-response-rows"] = (function() {
   var plugin = {};

   jsPsych.pluginAPI.registerPreload('audio-button-response-rows', 'stimulus', 'audio');

   plugin.info = {
      name: 'audio-button-response-rows',
etc.
```

In the long run I would like to move the prompt so that it displays above the buttons, since that is how I've made it in my normal audio-button trials, but that is not the source of the problem since I get the same error message even using the default prompt below buttons.

**Reply by jodeleeuw** [reply]
I think the issue is the `id` attribute of the code. It looks like you copied it over from the `html-button-response` plugin. Try changing `id="jspsych-html-button-response-button-'` to `id="jspsych-audio-button-response-button-'`.
